crates/gossip/src/initiate.rs

```diff
@@ -0,0 +1,646 @@
+
+        let url = select_next_target(
+            Duration::from_secs(60),
+            harness.peer_store.clone(),
+            harness.local_agent_store.clone(),
+            harness.peer_meta_store.clone(),
+        )
+        .await
+        .unwrap();
+
+        assert_eq!(None, url);
+    }
+
+    #[tokio::test]
+    async fn prioritise_never_gossiped_peers() {
+        enable_tracing();
+
+        let harness = Harness::create().await;
+
+        harness.new_local_agent(DhtArc::FULL).await;
+
+        let agent_2 = harness
+            .new_remote_agent(
+                Some(Url::from_str("ws://test:80/2").unwrap()),
+                Some(DhtArc::FULL),
+                Some(DhtArc::FULL),
+            )
+            .await;
+        let agent_3 = harness
+            .new_remote_agent(
+                Some(Url::from_str("ws://test:80/3").unwrap()),
+                Some(DhtArc::FULL),
+                Some(DhtArc::FULL),
+            )
+            .await;
+
+        let min_interval = K2GossipConfig::default().min_initiate_interval();
+
+        // Mark that we've gossiped with agent_3, but not recently
+        harness
+            .peer_meta_store
+            .set_last_gossip_timestamp(
+                agent_3.url.clone().unwrap(),
+                (Timestamp::now()
+                    - Duration::from_secs(min_interval.as_secs() + 90))
+                .unwrap(),
+            )
+            .await
+            .unwrap();
+
+        let url = select_next_target(
+            min_interval,
+            harness.peer_store.clone(),
+            harness.local_agent_store.clone(),
+            harness.peer_meta_store.clone(),
+        )
+        .await
+        .unwrap();
+
+        assert_eq!(agent_2.url.clone().unwrap(), url.unwrap());
+    }
+
+    #[tokio::test]
+    async fn randomly_select_from_multiple_new_peers() {
+        enable_tracing();
+
+        let harness = Harness::create().await;
+
+        harness.new_local_agent(DhtArc::FULL).await;
+
+        harness
+            .new_remote_agent(
+                Some(Url::from_str("ws://test:80/2").unwrap()),
+                Some(DhtArc::FULL),
+                Some(DhtArc::FULL),
+            )
+            .await;
+        harness
+            .new_remote_agent(
+                Some(Url::from_str("ws://test:80/3").unwrap()),
+                Some(DhtArc::FULL),
+                Some(DhtArc::FULL),
+            )
+            .await;
+
+        let mut seen = HashSet::new();
+        for _ in 0..10 {
+            let url = select_next_target(
+                Duration::from_secs(60),
+                harness.peer_store.clone(),
+                harness.local_agent_store.clone(),
+                harness.peer_meta_store.clone(),
+            )
+            .await
+            .unwrap();
+
+            seen.insert(url.unwrap());
+        }
+
+        assert_eq!(2, seen.len());
+    }
+
+    #[tokio::test]
+    async fn prioritises_overlapping_arc() {
```

Comment:
```suggestion
    async fn selects_non_overlapping_when_no_overlapping_peers_available() {
```

Reply:
I was about to suggest to add another peer that does have the overlapping arc, but then realized that the non-overlapping peers will never be selected then, regardless of timestamp.

Reply:
Yeah that was the original intention of this test which is why it's confusingly named :)

Reply:
Renamed back, in light of the change below, this is now relevant again